docs: update ffi/README.md to include configs, metrics, and logs (#1111)
ava-labs/firewood: ffi/README.md

## Configuration

### Database Config

A `Config` should be provided when creating the database. A default config is provided at `ffi.DefaultConfig()`:

```go
&Config{
    NodeCacheEntries:     1000000,
    FreeListCacheEntries: 40000,
    Revisions:            100,
    ReadCacheStrategy:    OnlyCacheWrites,
}
```

If no config is provided (`config == nil`), the default config is used. A description of all available values, and the default if not set, is available below.

#### `Truncate` - `bool`

If set to `true`, an empty database will be created, overriding any existing file. Otherwise, if a file exists, that file will be loaded to create the database. In either case, if the file doesn't exist, it will be created.

*Default*: `false`

#### `Revisions` - `uint`

Indicates the number of committed roots accessible before the diff layer is compressed. Must be explicitly set if the config is specified to at least 2.

#### `ReadCacheStrategy` - `uint`

> Should be one of `OnlyCacheWrites`, `CacheBranchReads`, or `CacheAllReads`. In the latter two cases, writes are still cached.

Reviewer: (nit): can we map the individual constants to their literal values (e.g. `OnlyCacheWrites = 1`, `CacheBranchReads = 2`, etc.)

*Default*: `OnlyCacheWrites`

#### `NodeCacheEntries`- `uint`

The number of nodes in the database that are stored in cache. Must be explicitly set if the config is supplied.

#### `FreeListCacheEntries` - `uint`

The number of entries in the free list (see [Firewood Overview](../README.md)). Must be explicitly set if the config is supplied.

### Metrics

By default, metrics are not enabled in Firewood's FFI. However, if compiled with this option, they can be recorded by a call to `StartMetrics()` or `StartMetricsWithExporter(port)`. One of these may be called exactly once, since it starts the metrics globally on the process.

To use these metrics, you can:

- Listen on the port specified, if you started the metrics with the exporter.
- Call `GatherMetrics()`, which returns an easily parsable string containing all metrics.
- Create the Prometheus gatherer, and call `Gather`. This can easily be integrated into other applications which already use prometehus. Example usage is below:

```go
gatherer := ffi.Gatherer{}

> gathering, err := gatherer.Gather()

Reviewer: ```suggestion
metrics, err := gatherer.Gather()
```
(nit/feel-free-to-ignore)

To support development in [Coreth](https://github.com/ava-labs/coreth), Firewood pushes static libraries for Ethereum-compatible hashing to [firewood-go-ethhash](https://github.com/ava-labs/firewood-go-ethhash) with `ethhash` enabled by default. To use Firewood's native hashing structure, you must still build the static library separately.

## Configuration

### Database Config

A `Config` must be provided when creating the database. A default config is provided at `ffi.DefaultConfig()`:

```go
&Config{
    NodeCacheEntries:     1000000,
    FreeListCacheEntries: 40000,
    Revisions:            100,
    ReadCacheStrategy:    OnlyCacheWrites,
}
```

A description of all available values, and the default if not set, is available below.

#### Truncate - `bool`

If set to `true`, an empty database will be created, overriding any existing file. Otherwise, if a file exists, that file will be loaded to cr:w
eate the database. In either case, if the file doesn't exist, it will be created.

*Default*: `false`

#### Revisions - `uint`

> Indicates the number of committed roots accessible before the diff layer is compressed. Must be explicitly set if the config is specified to at least 2.

Reviewer 2: ```suggestion
Indicates the minimum number of committed roots accessible before the they expire. If set, it must be eat least 2.
```

Reviewer 2: What happens if it's not set?

Author: 0 - returns an error.

#### ReadCacheStrategy - `uint`

> Should be one of `OnlyCacheWrites`, `CacheBrancheReads`, or `CacheAllReads`. In the latter two cases, writes are still cached.

Reviewer 2: typo. Does golang doc have some way to avoid these?
```suggestion
Should be one of `OnlyCacheWrites`, `CacheBranchReads`, or `CacheAllReads`. In the latter two cases, writes are still cached.
```

*Default*: `OnlyCacheWrites`

#### NodeCacheEntries

> The number of nodes in the database that are stored in cache. Must be explicitly set if the config is supplied.

Reviewer 2: What's the default if you don't set it?

#### FreeListCacheEntries

The number of entries of the free list (see [Firewood Overview](../README.md)). Must be explicitly set if the config is supplied.

### Metrics

By default, metrics are not enabled in Firewood's FFI. However, if compiled with this option, they can be recorded by a call to `StartMetrics()` or `StartMetricsWithExporter(port)`. One of these may be called exactly once, since it starts the metrics globally on the process.

To use these metrics, you can:

- Listen on the port specified, if you started the metrics with the exporter.
- Call `GatherMetrics()`, which returns an easily parsable string containing all metrics.
- Create the Prometheus gatherer, and call `Gather`. This can easily be integrated into other applications which already use prometehus. Example usage is below:

```go
gatherer := ffi.Gatherer{}
gathering, err := gatherer.Gather()
```

### Logs

> Logs are configured globally on the process, and not enabled by default. They can be enabled using the `StartLogs(config)` function. Firewood must be built with the `logger` feature for this function to work. If a config isn't provided, the default values will be used.

Reviewer 2: nit: wordy
```suggestion
Logs are configured globally on the process, and not enabled by default. They can be enabled using the `StartLogs(config)` function. Firewood must be built with the `logger` feature for this function to work. This should be called before opening the database.
```

#### Path

The path to the file where the logs will be written.

*Default*: `{TEMP}/firewood-log.txt`, where `{TEMP}` is the platform's temporary directory. For Unix-based OSes, this is typically `/tmp`.

#### FilterLevel

> One of `trace`, `debug`, `info`.

Reviewer 2: What about 'warn' and 'error'?
